### PR TITLE
docs(advanced): Improve the Testing Form section

### PR DESCRIPTION
**Pull request by Belco90**
Hi!

[eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) author here. I've found many people following the wrong advice from the `react-hook-form` docs about how to fix the React Testing Library `act()` warning.

I've updated the docs and the examples, so it's on the same page as the official Testing Library best practices.

### CODE REVIEW

**Review by bluebill1049 — APPROVED** [review]
thanks very much for correct this 🙏

File: src/components/codeExamples/testingForm.ts [hunk 1]
```diff
@@ -54,82 +54,85 @@ const mockLogin = jest.fn((email, password) => {
   return Promise.resolve({ email, password });
 });
 
-describe("App", () => {
```
Comment (Belco90) [inline src/components/codeExamples/testingForm.ts:57]: [Avoid nesting when you are testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing): no redundant `describe` blocks

File: src/components/codeExamples/testingForm.ts [hunk 2]
```diff
@@ -54,82 +54,85 @@ const mockLogin = jest.fn((email, password) => {
...
-  beforeEach(() => {
-    render(<App login={mockLogin} />);
```
Comment (Belco90) [inline src/components/codeExamples/testingForm.ts:59]: [Avoid nesting when you are testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing): inline renders

File: src/components/codeExamples/testingForm.ts [hunk 3]
```diff
@@ -54,82 +54,85 @@ const mockLogin = jest.fn((email, password) => {
...
+    }
+  });
 
-    expect(await screen.findAllByRole("alert")).toHaveLength(2);
-    expect(mockLogin).not.toBeCalled();
+  fireEvent.submit(screen.getByRole("button"));
+
+  expect(await screen.findAllByRole("alert")).toHaveLength(1);
+  expect(mockLogin).not.toBeCalled();
+  expect(screen.getByRole("textbox", { name: /email/i })).toHaveValue("test");
```
Comment (Belco90) [inline src/components/codeExamples/testingForm.ts:86]: Using `toHaveValue` matcher.

File: src/components/codeExamples/testingForm.ts [hunk 4]
```diff
@@ -159,29 +162,29 @@ export default function App() {
...
 import React from "react";
-import { render, screen, act } from "@testing-library/react";
+import { render, screen } from "@testing-library/react";
 import App from "./App";
 
-describe("App", () => {
-  it("should have a submit button", async () => {
-    await act(async () => {
-      render(<App />)
-    });
```
Comment (Belco90) [inline src/components/codeExamples/testingForm.ts:182]: The main reason why I created this PR: [this is a common mistake](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily).

**Comment by Belco90** [comment]
That was quick, thanks!

**Comment by bluebill1049** [comment]
thank you @Belco90 🙏

**Pull request merged**